File: 2trabajoConstructores/js/clase abstracta/AnimalCopia.js
const Animal = require("./Animal.js");

class AnimalCopia extends Animal {
    constructor(otroAnimal) {
        super();
        this.nombre = otroAnimal.nombre;
        this.sonido = otroAnimal.sonido;
    }

    hacerSonido() {
        console.log(`${this.nombre} (copia) hace: ${this.sonido}`);
    }
}

module.exports = AnimalCopia;

File: 2trabajoConstructores/c#/claseAbstracta/AnimalCopia.cs
public class AnimalCopia : AnimalBase
{
    public AnimalCopia(AnimalBase otro)
    {
        Nombre = otro.Nombre;
        Sonido = otro.Sonido;
    }

    public override void HacerSonido()
    {
        Console.WriteLine($"{Nombre} hace: {Sonido}");
    }
}
